tests/test_usecase.py

```diff
@@ -29,7 +29,7 @@
 CAPTURE_OUTPUT = True
 
 
-class FakeSystemExit(Exception):
+class FakeSystemExit(SystemExit):
```

Comment:
Is this needed? Let's not start a philosophical debate about whether or not a `FakeSystemExit` is a `SystemExit` but in the new version it would not be cached by an `except Exception` which is probably not wanted (although `unittest` seems to handle that correctly).

Reply:
Base `Exceptions` don't have the `code` attribute like `SystemExit`s do.  I can change it to subclass `Exception` and redefine `__init__`.  This just seemed simpler and didn't break the unittests (for now at least).

Reply:
Probably remove this.

Reply:
Sorry, my message was written before yours.

You're probably right. At the end both versions are fine but let's go for the simplest. I guess any test suite would catch `BaseException` and not just `Exception` anyway.